analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/Verifier.cs

```diff
@@ -176,6 +182,11 @@ private IEnumerable<Compilation> CompileRazor()
                 MSBuildLocator.RegisterDefaults();
             }
 
+            if (!razorSupportedFrameworks.Contains(builder.Framework))
+            {
+                throw new InvalidOperationException("Razor compilation is supported starting from .NET 6 and onwards.");
```

Comment:
I would explicitly display the list of the supported framework in the message.
As it is now, it implies that `net8.0` is supported and might give us some headaches in the future.